### PR TITLE
[MRG+1] lazy-load s3 boto

### PR DESCRIPTION
**Pull request by nyov**
companion to #1357
removes s3/boto dependency on `optional_features`

### CODE REVIEW

File: tests/test_downloader_handlers.py [hunk 1]
```diff
@@ -372,7 +372,7 @@ def _test(response):
         return self.download_request(request, Spider('foo')).addCallback(_test)
 
 
-class DeprecatedHttpProxyTestCase(unittest.TestCase):
+class DeprecatedHttpProxyTestCase(HttpProxyTestCase):
```
Comment (kmike) [inline tests/test_downloader_handlers.py]: is this change related?

Comment (nyov) [inline tests/test_downloader_handlers.py]: No. I thought I would fix it while I was there.

File: scrapy/core/downloader/handlers/s3.py [hunk 2]
```diff
@@ -1,41 +1,40 @@
...
-    _S3Connection = _v20_S3Connection
-
 
 class S3DownloadHandler(object):
 
     def __init__(self, settings, aws_access_key_id=None, aws_secret_access_key=None, \
             httpdownloadhandler=HTTPDownloadHandler):
-        if 'boto' not in optional_features:
+
+        try:
```
Comment (dangra) [inline scrapy/core/downloader/handlers/s3.py]: can I ask to group all this imports into a function or method? it returns _S3Connection

**Comment by nyov** [comment]
> can I ask to group all this imports into a function or method? it returns _S3Connection

I hope this looks better, if not let me know :)
